libs/dotnet-sdk/Registration/RegisterFeatureBoard.cs

```diff
@@ -11,32 +12,45 @@ namespace FeatureBoard.DotnetSdk.Registration;
 public static class RegisterFeatureBoard
 {
   private static UpdateStrategy _updateStrategy = UpdateStrategy.None;
-  public static FeatureBoardBuilder AddFeatureBoard<TFeatures, TAudienceProvider>(this IServiceCollection services) where TFeatures : class, IFeatures where TAudienceProvider : class, IAudienceProvider
+
+  private static EntityTagHeaderValue? lastETag = null;
+
+  public static FeatureBoardBuilder AddFeatureBoard<TFeatures>(this IServiceCollection services, IConfigurationRoot configuration) where TFeatures : class, IFeatures
   {
-    var configuration = services.BuildServiceProvider().GetRequiredService<IConfiguration>();
     services.AddOptions<FeatureBoardOptions>().Bind(configuration.GetSection(nameof(FeatureBoardOptions))).ValidateDataAnnotations();
 
-    services.AddHttpClient<IFeatureBoardHttpClient, FeatureBoardHttpClient>().ConfigureHttpClient((serviceProvider, client) =>
+    services.AddHttpClient<IFeatureBoardHttpClient, FeatureBoardHttpClient>().ConfigureHttpClient(static (serviceProvider, client) =>
     {
       var options = serviceProvider.GetRequiredService<IOptions<FeatureBoardOptions>>();
       client.BaseAddress = options.Value.HttpEndpoint;
       client.DefaultRequestHeaders.Add("x-environment-key", options.Value.EnvironmentApiKey);
-      client.Timeout = options.Value.MaxAge - TimeSpan.FromMilliseconds(3); //prevent multiple requests running at the same time.
+      // client.Timeout = options.Value.MaxAge - TimeSpan.FromMilliseconds(3); //prevent multiple requests running at the same time.
```

Comment:
This supposed to be commented out?

Reply:
Ah! I meant to initiate a comment on this when I put up the PR... I feel like it is not needed - the way the `PeriodicTimer` in `PollingUpdateStrategyBackgroundService` works means that there will never be multiple requests running at same time as far as I can tell. And the semaphore in `FeatureBoardService` also ensures there will never be multiple requests in flight with that strategy too. Is there something I'm missing?